File: scripts/prepare_resume_author_payload.py
import argparse, json
from pathlib import Path


def read_json(path: Path, default):
    if path.exists():
        return json.loads(path.read_text(encoding='utf-8'))
    return default


def main():
    ap = argparse.ArgumentParser()
    ap.add_argument('--run-dir', required=True)
    ap.add_argument('--cycle', required=True, type=int)
    args = ap.parse_args()

    run = Path(args.run_dir)
    payload = {
        'job_description': read_json(run / 'job_description.json', {}),
        'candidate_master_profile': read_json(run / 'candidate_master_profile.json', {}),
        'match_analysis_md': (run / 'match_analysis.md').read_text(encoding='utf-8') if (run / 'match_analysis.md').exists() else '',
        'review_feedback': read_json(run / f'review_feedback_cycle_{args.cycle-1}.json', {}) if args.cycle > 1 else None,
    }

    out = run / 'resume_author_payload.json'
    out.write_text(json.dumps(payload, ensure_ascii=False, indent=2), encoding='utf-8')
    print(json.dumps({'resume_author_payload': str(out)}))


if __name__ == '__main__':
    main()

File: scripts/prepare_resume_author_payload.mjs
import fs from 'node:fs';
import path from 'node:path';
const args = Object.fromEntries(process.argv.slice(2).map((v,i,a)=>v.startsWith('--')?[v.slice(2),a[i+1]]:null).filter(Boolean));
const runDir=args['run-dir'];
const cycle=Number(args['cycle']||1);
function readJson(p,d){try{return JSON.parse(fs.readFileSync(p,'utf8'));}catch{return d;}}
const payload={
  job_description: readJson(path.join(runDir,'job_description.json'),{}),
  candidate_master_profile: readJson(path.join(runDir,'candidate_master_profile.json'),{}),
  match_analysis_md: fs.existsSync(path.join(runDir,'match_analysis.md'))?fs.readFileSync(path.join(runDir,'match_analysis.md'),'utf8'):'',
  review_feedback: cycle>1?readJson(path.join(runDir,`review_feedback_cycle_${cycle-1}.json`),{}):null
};
const out=path.join(runDir,'resume_author_payload.json');
fs.writeFileSync(out, JSON.stringify(payload,null,2),'utf8');
process.stdout.write(JSON.stringify({resume_author_payload:out}));